armi/bookkeeping/db/database3.py

```diff
@@ -1611,8 +1612,16 @@ def packSpecialData(
 
     """
 
+    # Check to make sure that we even need to do this. If the numpy data type is
+    # not "O", chances are we have nice, clean data.
+    if data.dtype != "O":
+        return data, {}
```

Comment:
slightly out of scope but the return values for this method aren't super well defined in the docstring.